code/datums/emote.dm

```diff
@@ -483,6 +486,9 @@
 		if(species_type_whitelist_typecache && H.dna && !is_type_in_typecache(H.dna.species, species_type_whitelist_typecache))
 			return FALSE
 
+		if(species_type_blacklist_typecache && H.dna && is_type_in_typecache(H.dna.species, species_type_blacklist_typecache))
+			return FALSE
```

Comment:
```suggestion
		if(H.dna)
				if(!is_type_in_typecache(H.dna.species, species_type_whitelist_typecache))
					return FALSE

				if(is_type_in_typecache(H.dna.species, species_type_blacklist_typecache))
					return FALSE
```
two things, might aswell put `H.dna` into its own check to reduce redundancy. Second, `is_type_in_typecache()` will already null check the provided list, so you don't need to do that yourself.

Reply:
good call. I'll probably add a ? operator on top of that since I'm pretty sure dna and/or species can be null

Reply:
Testing it a bit, I think we actually do need to check for the typecache. It looks like it returns false if the typecache doesn't exist, which means that if we don't have a whitelist, species won't be able to use the emote.